### PR TITLE
added window webContents event

### PR DESCRIPTION
**Pull request by ZEROKISEKI**
added astilectron write event when electron listen "did-get-redirect-request" and "will-navigate" event. If this merge and release a new version(such as v.0.16.0), maybe `go-astilectron` have to change the const `VersionAstilectron` to `0.16.0`.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -364,6 +364,16 @@ function windowCreate(json) {
         sessionCreate(elements[json.targetID].webContents, json.sessionId)
         client.write(json.targetID, consts.eventNames.windowEventDidFinishLoad)
     })
+    elements[json.targetID].webContents.on('did-get-redirect-request', (event, oldUrl, newUrl) => {
```
Comment (asticode) [inline main.js]: I'd rather you send both `oldUrl` and `newUrl` in the payload.
That means you'll have to add them as well in the [GO event](https://github.com/asticode/go-astilectron/blob/master/event.go#L32)

File: main.js [hunk 2]
```diff
@@ -364,6 +364,16 @@ function windowCreate(json) {
...
+        client.write(json.targetID, consts.eventNames.windowEventDidGetRedirectRequest, {
+            url: newUrl
+        })
+    })
+    elements[json.targetID].webContents.on('will-navigate', (event, url) => {
+        client.write(json.targetID, consts.eventNames.windowEventWillNavigate, {
+            url
```
Comment (asticode) [inline main.js]: I'm not familiar with that notation. Does writing `url` on its own means the payload will be the following (if the value of url was `https://github.com`): `{"url":"https://github.com"}` ?

**Review by asticode — CHANGES_REQUESTED** [review]
The changes in this project seems fine.

Just a minor change though.

File: main.js (hunk 2 above)
Comment (ZEROKISEKI) [reply main.js]: the `{ url }` will parse to `{ "url" : url }`

**Comment by ZEROKISEKI** [comment]
@asticode, when trigger `"did-get-redirect-request"` event it will have the payload `{ "oldUrl": "xxx", "newUrl": "xxx" }`, trigger `"will-navigate"` will have the payload `{ "url": "xxx" }`, I'm no sure how to modify the [Go Event struct](https://github.com/asticode/go-astilectron/blob/master/event.go#L32) properly, could you give some suggestions about it?

**Comment by asticode** [comment]
1) There might be something I'm misunderstanding, but in your PR for the event `did-get-redirect-request` I see `{url: newUrl}`. I don't see how this will become `{ "oldUrl": "xxx", "newUrl": "xxx" }`...
2) To modify properly the Event Go struct, you have to add the following in the struct **in alphabetical order**:

```go
URLNew string `json:"newUrl,omitempty"`
URLOld string `json:"oldUrl,omitempty"`
```

**Comment by asticode** [comment]
@ZEROKISEKI I'm sorry I got mixed up and I had to push force on this repository :( The content of your PR is fine though.

Could you rebase on master and resubmit your PR?

Sorry for the inconvenience.

**Comment by ZEROKISEKI** [comment]
@asticode Does it means I haved to fork again and create a new PR 😂 😂? I'm not very similar to the operation "rebase" 😂 😂

**Comment by asticode** [comment]
You should only have to `git pull --rebase git@github.com:asticode/astilectron.git master` and create a new PR.

Once again, I'm sorry for the inconvenience, that's my bad, I got mixed up :(

**Pull request closed without merging**